### ISSUE TITLE
list of llms (for code)

### ISSUE THREAD

**Issue by andre15silva** [ASSERT-KTH/llm-repair-them-all#49]

**Comment by monperrus** [comment]
aiXcoder-7B: A Lightweight and Effective Large Language Model for Code Completion
https://www.semanticscholar.org/reader/2c5dd0f56eff1caa3edb20354374a9585181ea73

**Comment by monperrus** [comment]
Tencent 's Hunyuan ([huggingface](https://huggingface.co/tencent/Tencent-Hunyuan-Large), [paper](https://arxiv.org/pdf/2411.02265))

**Comment by monperrus** [comment]
update from https://github.com/underlines/awesome-ml/

| Model        | Link                                                                 | Description                                                                      |
|--------------|----------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| CursorCore | https://huggingface.co/collections/TechxGenus/cursorcore-series-6706618c38598468866b60e2 | Coding LLMs for use within [CursorCore](https://github.com/TechxGenus/CursorCore) and the VL-Vicuna model is a novel VL-LLM. [Paper](https://arxiv.org/abs/2305.01278), [code](https://github.com/VPGTrans/VPGTrans) |
| gemma2 2b | https://huggingface.co/bartowski/gemma-2-2b-it-GGUF | 2b small language models, trained on 2T tokens, 87% code 13% English / Chinese, up to 33B with 16K context sizemodel by google achieving SOTA performance for sub 3b models on coding benchmarks |
| DeepSeekCoderv2 | https://github.com/deepseek-ai/DeepSeek-Coder-V2?tab=readme-ov-file#2-model-downloads | 16b and code, [news](https://joinup.ec.europa.eu/collection/open-source-observatory-osor/news/new-open-source-ai-model-poro-challenges-french-mistral)|
| codegemma | https://huggingface.co/google/codegemma-7b | google's coding models from 2b base, 7b base and 52B MoE at 12B each|
| grantie | https://huggingface.co/collections/ibm-granite/granite-code-models-6624c5cec322e4c148c8b330 | IBMs code models available in 3b, 8b, 20b size as base and tool use supporting Embedinstruct variants with up to 128k context size|
| codeqwen1.5 | https://huggingface.co/Qwen/CodeQwen1.5-7B | base and Rerank methodology. [model weights](https://huggingface.co/CohereForAI/c4ai-command-r-v01)|
| InternLM2.5 | https://huggingface.co/internlm/internlm2_5-7b-chat | 7B base andinstruct models for code completion and chat - [DeepSeek-V2](https://github.com/deepseek-ai/DeepSeek-V2#2-model-downloads) 21B Strong, Economical,models focusing reasoning, math and Efficient Mixture-of-Experts Language Model|
| CodeGeeX4 | https://huggingface.co/THUDM/codegeex4-all-9b | 9B multilingual code generation model based on phi-3-mini, allowing tofor chat and instruct based onwith ajson template that the model fills from unstructured text provided|
| Mamba-Codestral | https://huggingface.co/mistralai/Mamba-Codestral-7B-v0.1 | by mistral based on the Mamba2 architecture performing on parquality of current SOTA closed source models|
| CodeStral-22B | https://huggingface.co/mistralai/Codestral-22B-v0.1 | Coding model trained on 80+ languages with great performanceinstruct and 128k context windows forFill in the 7 and 72b models|
| Granite | https://huggingface.co/ibm-granite | family of Code Models from IBM with7B parameters and good quality|
| wavecoder-ultra-6.7b | https://huggingface.co/microsoft/wavecoder-ultra-6.7b | covering four general code-related tasks: code generation, code summary, code translation, and 236b mixturecode repair|
| aiXcoder | https://huggingface.co/aiXcoder/aixcoder-7b-base | 7B Code LLM for code completion, comprehension, generation|
| StarCoder2 | https://huggingface.co/bigcode/starcoder2-15b | 15B, 7B and 3B code completion models trained on The Stack v2|
| Poro | https://huggingface.co/LumiOpen/Poro-34B | SiloGen model checkpoints of a family of multilingual open source LLMs covering all official European languages and code, [news](https://joinup.ec.europa.eu/collection/open-source-observatory-osor/news/new-open-source-ai-model-poro-challenges-french-mistral)|
| deepseek-coder | https://github.com/deepseek-ai/DeepSeek-Coder | code language models, trained on 2T tokens, 87% code 13% English / Chinese, up to convert HTML33B with 16K context size achieving SOTA performance on coding benchmarks|
| CodeShell | https://github.com/WisdomShell/codeshell/blob/main/README_EN.md | code LLM with 7b parameters trained on 500b tokens, context length of 8k outperforming CodeLlama and Starcoder on humaneval, [weights](https://huggingface.co/WisdomShell/CodeShell)|
| salesforce/CodeT5 | https://github.com/salesforce/codet5 | code assistant, has released their [codet5+ 16b](https://huggingface.co/Salesforce/codet5p-16b) and content identification an LLM task|
| replit-code | https://huggingface.co/replit/ | focused on Code Completion. The model has been trained on a subset of the [Stack Dedup v1.2](https://arxiv.org/abs/2211.15533) dataset.|
| BigCode | https://huggingface.co/bigcode | Open Scientific collaboration to have native "listening" ability, using an early fusion technique,train a coding LLM | 
| CodeGeeX 13B | https://huggingface.co/spaces/THUDM/CodeGeeX| Multi Language Code Generation Model |

**Comment by monperrus** [comment]
DeepSeek-R1
DeepSeek-R1, A Transparent Challenger to OpenAI o1
https://www.deeplearning.ai/the-batch/deepseek-r1-a-transparent-challenger-to-openai-o1/
https://huggingface.co/deepseek-ai/DeepSeek-R1

will have it in repairbench as soon as it is served on openrouter

**Comment by monperrus** [comment]
GLM series from Tsinghua University and Zhipu AI includes the latest GLM3 and GLM4 models

Paper: ChatGLM: A family of large language models from GLM-130B to GLM-4 all tools. arXiv 2406.12793 (2023)

HF: https://huggingface.co/collections/THUDM/glm-4-665fcf188c414b03c2f7e3b7

**Comment by monperrus** [comment]
Model news:
[Gemma 3 27B](https://blog.google/technology/developers/gemma-3/) open weight models by Google
[AI21: Jamba 1.6 Large](https://huggingface.co/ai21labs/AI21-Jamba-Large-1.6), ￼￼
[Cohere Command A](https://cohere.com/blog/command-a)

**Comment by monperrus** [comment]
OpenHands LM v0.1 is a 32B open-source coding model fine-tuned from Qwen2.5-Coder-32B-Instruct using reinforcement learning techniques outlined in SWE-Gym.

- https://huggingface.co/all-hands/openhands-lm-32b-v0.1 
- https://openrouter.ai/all-hands/openhands-lm-32b-v0.1

**Comment by monperrus** [comment]
GLM-4-0414 Model Series
https://github.com/THUDM/GLM-4

**Comment by monperrus** [comment]
delivery from underlines/awesome-ml

| Model | Link | Description | Date added |
|---|---|---|---|
| OpenHands-LM | https://huggingface.co/all-hands | openhands 1.5b, 7b and 32b coding models with verified strong performance on SWE-Bench using the OpenHands Coding-Agent | 2025-04-25 |
| OpenThinker2-32B | https://huggingface.co/open-thoughts/OpenThinker2-32B | fine-tuned version of Qwen2.5-32B-Instruct on the OpenThoughts2-1M dataset with increased quality compared to base model | 2025-04-25 |
| Cogito-V1 | https://huggingface.co/collections/deepcogito/cogito-v1-preview-67eb105721081abe4ce2ee53 | Cogito model family with Qwen and Llama fine-tunes using Iterated Distillation and Amplification (IDA) to increase coding, STEM and IF quality compared to their base models | 2025-04-25 |
| DeepCoder-14B-Preview | https://huggingface.co/agentica-org/DeepCoder-14B-Preview | code reasoning LLM fine-tuned from DeepSeek-R1-Distilled-Qwen-14B using RL to scale up to long context lengths using DeepScaleR and GRPO+ | 2025-04-25 |
| ZR1-1.5B | https://huggingface.co/Zyphra/ZR1-1.5B | Fine Tuned DeepSeek-R1-Distill-Qwen-1.5B trained extensively on both verified coding and mathematics problems with reinforcement learning | 2025-04-25 |
| Skywork-OR1 | https://huggingface.co/Skywork/Skywork-OR1-32B-Preview | 7B, 7B math and 32B reasoning models with open sourced weights, training data and training code | 2025-04-25 |
| GLM-4 | https://huggingface.co/collections/THUDM/glm-4-0414-67f3cbcb34dd9d252707cb2e | 3 32B models as general, reasoning and deep reasoning variant as well as a 9B SML | 2025-04-25 |
| MAI-DS-R1 | https://huggingface.co/microsoft/MAI-DS-R1 | post-trained DeepSeek-R1 reasoning model by Microsoft AI that enhances responsiveness on blocked topics while maintaining strong reasoning capabilities | 2025-04-25 |
| Gemma 3 QAT | https://huggingface.co/google/gemma-3-27b-it-qat-q4_0-gguf | Quantization Aware Training models from Google, regaining bf16 quality in int4 quants and slashing memory footprint | 2025-04-25 ||---|---|---|---|---|

**Comment by monperrus** [comment]
ByteDance Seed-Coder

> we introduce Seed-Coder, a series of open-source LLMs comprising base, instruct and reasoning models of 8B size, minimizing human involvement in data construction

https://arxiv.org/abs/2506.03524
https://huggingface.co/ByteDance-Seed/Seed-Coder-8B-Instruct

**Comment by monperrus** [comment]
Yi-Coder is a series of open-source code language models that delivers state-of-the-art coding performance with fewer than 10 billion 
https://github.com/01-ai/Yi-Coder
https://huggingface.co/01-ai/Yi-Coder-9B

**Comment by monperrus** [comment]
the Qwen3-Coder family 
https://huggingface.co/collections/Qwen/qwen3-coder-687fc861e53c939e52d52d10
